### PR TITLE
feat: 1529 spar service dependencies status page

### LINKED ISSUE #1529: SPAR Service Dependencies Status Page
SPAR currently has four API dependencies. If any of them fail, a significant portion of SPAR's features will not function.

Although SPAR provides surface-level error displays, users may not understand the root cause of the issue and might assume the failure is due to SPAR itself, even though it could be related to an external factor, such as the Forest Client API being down.

Task

Build a status page akin to Github Status

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Closes #1529

New route: /service-status
The "Return to main page" only appears when users are logged in

Screenshot 2024-08-21 at 5 11 17 PM

Screenshot 2024-08-21 at 5 10 53 PM

Screenshot 2024-08-21 at 5 11 32 PM

#### New
- a service status page for SPAR dependencies. 

#### Changed  
- Layout component to display an error for users that are logged in
- streamlined App.tsx

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-39-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1539-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1539-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by DerekRoberts — APPROVED** [review]
Very nice! :)

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -23,32 +23,57 @@ import FourOhFour from './views/FourOhFour';
...
-  const MAX_RETRIES = 3;
-
   const { signed, isCurrentAuthUser, selectedClientRoles } = useContext(AuthContext);
 
   useEffect(() => {
     isCurrentAuthUser(window.location.pathname);
   }, []);
 
-  const roleSelectionRouter = createBrowserRouter([
+  const roleSelectionRoutes: RouteObject[] = [
```
Comment (RMCampos) [inline frontend/src/App.tsx:69]: praise: nice 👍🏻 (data types)

**Review by RMCampos — APPROVED** [review]
Great job!

File: frontend/src/views/ServiceStatus/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    kind="warning"
+                    statusIconDescription="notification"
+                    subtitle={
+                      `${statusQueries.filter((query) => query.status === 'error').length} out of
+                       ${SPAR_DEPENDENCIES.length} service${SPAR_DEPENDENCIES.length > 0 ? 's' : ''}
+                       are down.
+                      `
+                    }
+                    /*
+                     * It's grammatically incorrect to use 'is' for singluar subject
```
Comment (mgaseta) [inline frontend/src/views/ServiceStatus/index.tsx:72]: Typo on  "singular"

File: frontend/src/views/ServiceStatus/styles.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      .status-notification {
+        max-width: none;
+      }
+
+      .bx--inline-notification__title {
+        font-weight: 600;
+      }
+    }
+
+
```
Comment (mgaseta) [inline frontend/src/views/ServiceStatus/styles.scss:24]: Remove empty lines.

**Review by mgaseta — APPROVED** [review]
Nice job! LGTM!
Left only 2 cosmetics markups.

File: frontend/src/views/ServiceStatus/index.tsx (hunk 2 above)
Comment (craigyu) [reply frontend/src/views/ServiceStatus/index.tsx:72]: a typo on a comment about grammar, how ironic 💩

**Pull request merged**